lib/file.js

```diff
@@ -147,6 +150,63 @@ class File extends EventEmitter {
     render.render(this, elem, opts, cb)
   }
 
+  _serve (req) {
+    const res = {
+      status: 200,
+      headers: {
+        // Support range-requests
+        'Accept-Ranges': 'bytes',
+        'Content-Type': this._getMimeType()
+      }
+    }
+    // force the browser to download the file if there is a specific header present
+    if (req.headers.get('upgrade-insecure-requests') === '1') {
+      res.headers['Content-Type'] = 'application/octet-stream'
+      res.headers['Content-Disposition'] = 'attachment'
+    }
+
+    // `rangeParser` returns an array of ranges, or an error code (number) if
+    // there was an error parsing the range.
+    let range = rangeParser(this.length, req.headers.get('range') || '')
+
+    if (Array.isArray(range)) {
+      res.status = 206 // indicates that range-request was understood
+
+      // no support for multi-range request, just use the first range
+      range = range[0]
+
+      res.headers['Content-Range'] = `bytes ${range.start}-${range.end}/${this.length}`
+      res.headers['Content-Length'] = `${range.end - range.start + 1}`
+    } else {
+      range = null
+      res.headers['Content-Length'] = this.length
+    }
+
+    res.headers['Cache-Control'] = 'no-cache, no-store, must-revalidate, max-age=0'
+    res.headers.Expires = '0'
+    res.body = req.method === 'HEAD' ? '' : 'STREAM'
+
+    // TODO: add support for passthrough of readStreams, so that the user can for example parse video data
+    return [res, req.method === 'GET' && this.createReadStream(range)]
+  }
+
+  getStreamURL (cb) {
+    if (typeof window === 'undefined') throw new Error('browser-only method')
+    if (!this._serviceWorker) throw new Error('No worker registered')
+    if (this._serviceWorker.state !== 'activated') throw new Error('Worker isn\'t activated')
+    const url = `${window.location.pathname.substr(0, window.location.pathname.lastIndexOf('/') + 1)}webtorrent/${this._torrent.infoHash}/${encodeURI(this.path)}`
+    cb(null, url)
+  }
+
+  streamTo (elem, cb) {
+    if (typeof window === 'undefined') throw new Error('browser-only method')
+    if (!this._serviceWorker) throw new Error('No worker registered')
+    if (this._serviceWorker.state !== 'activated') throw new Error('Worker isn\'t activated')
+    // TODO: fix this \/
```

Comment:
Please go deeper in detail as what there is to fix

Reply:
oh, when I was first writing it I didin't bother checking if there are any HTML elements which load their content in a way different than using `src` but then realized all do so that can be removed